corehq/apps/users/views/mobile/users.py

```diff
@@ -756,11 +761,47 @@ def create_mobile_worker(self, in_data):
             phone_number = self.new_mobile_worker_form.cleaned_data['phone_number']
             couch_user.set_default_phone_number(phone_number)
             send_account_confirmation_sms_if_necessary(couch_user)
+
+        plan_limit, user_count = Subscription.get_plan_and_user_count_by_domain(self.domain)
+        self.check_and_send_limit_email(self.domain, plan_limit, user_count, user_count - 1)
         return {
             'success': True,
             'user_id': couch_user.userID,
         }
 
+    @staticmethod
+    def check_and_send_limit_email(domain, plan_limit, user_count, prev_count):
+        ENTERPRISE_LIMIT = -1
+        if plan_limit == ENTERPRISE_LIMIT:
+            return None
+
+        WARNING_PERCENT = 0.9
+        if user_count >= plan_limit > prev_count:
+            at_capacity = True
+        elif plan_limit > user_count >= (WARNING_PERCENT * plan_limit) > prev_count:
+            at_capacity = False
+        else:
+            return None
+
+        billing_admins = [admin.username for admin in WebUser.get_billing_admins_by_domain(domain)]
+        admins = [admin.username for admin in WebUser.get_admins_by_domain(domain)]
+
+        if at_capacity:
+            subject = _("User count has reached the Plan limit for {}").format(domain)
+        else:
+            subject = _("User count has reached 90% of the Plan limit for {}").format(domain)
+        send_html_email_async(
+            subject,
+            admins + billing_admins,
```

Comment:
I don't know how well this will handle translations if the admins don't share the same language as whoever called the `create_mobile_worker` function. `subject` will be translated to whatever language the caller has set. If this is a concern, you could look at something like [the report notification code](https://github.com/dimagi/commcare-hq/blob/7ac73b311c325fa3824dc8b2d4c32e3e7b784526/corehq/apps/saved_reports/models.py#L739-L741), which is one example I'm aware of where we localize the emails to the recipient.

Reply:
The function you linked (or more specifically `recipients_by_language`) sounds like something that could be reworked to be more flexibly applied to different areas so I'll make a note to come back to this as it's own thing. But I don't really think different translations will be a problem here since the emails in this scenario are limited to a singular project's admins (who I presume work together).

Reply:
I'm not sure of the scope of the problem. I overlooked that both the `create_mobile_worker` function and `import_users` will hit these translations. While it's likely that the language for the user who triggers `create_mobile_worker` is shared by the admins, I'm less sure that tasks inherit the language settings of their caller. If not, then you could have a project where users and admins use something like Swahili receiving an email in English. [Possibly relevant celery link?](https://stackoverflow.com/questions/7199256/translation-in-celery-tasks#comment58178158_7208163). I don't think this should block this work, but its something to keep in mind.

Reply:
Yea I think you’re right needing to explicitly reference the caller/user’s language in order for the translations to take effect but I’m having a hard time finding other examples in the code, since I assume there’s at least one or two other instances where tasks trigger emails with translated strings. But in the interest of time I’m gonna lump this together with the previous note I left above and come back to both when I have more time!